<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Robot name parameter -->
    <arg name="robot_name" default="locobot" />
    
    <!-- Topic parameters -->
    <arg name="object_marker_topic" default="/$(arg robot_name)/detected_object/marker" />
    <arg name="bbox_depth_topic" default="/$(arg robot_name)/camera/yolo/bbox_depth" />
    
    <!-- Rviz Param -->
    <arg name="rviz_frame" default="map" />
    <arg name="rvizconfig" default="$(find yolo_vision)/rviz/lips_ws/src/yolo_vision/rviz/xslocobot_description_yolo.rviz" />
    <arg name="use_rviz" default="true" doc="Set to false to disable RViz visualization" />
    
    <!-- Launch the navigation controller that approacher depends on -->
    <include file="$(find object_scout)/launch/navigation.launch">
        <arg name="robot_name" value="$(arg robot_name)" />
    </include>
    
    <!-- Include the object mapper launch file from yolo_vision -->
    <include file="$(find yolo_vision)/launch/object_mapper.launch">
        <arg name="robot_name" value="$(arg robot_name)"/>
        <arg name="rviz_frame" value="$(arg rviz_frame)"/>
        <arg name="rvizconfig" value="$(arg rvizconfig)"/>
        <arg name="object_marker_topic" value="$(arg object_marker_topic)"/>
        <arg name="bbox_depth_topic" value="$(arg bbox_depth_topic)"/>
        <arg name="use_rviz" value="false"/>
    </include>
    
    <!-- Launch the object approacher node -->
    <node name="object_approacher" pkg="object_scout" type="approacher_node.py" output="screen">
        <param name="robot_name" value="$(arg robot_name)" />
        <param name="bbox_depth_topic" value="$(arg bbox_depth_topic)" />
    </node>
    
    <!-- RViz (conditionally launched) -->
    <group if="$(arg use_rviz)">
        <node name="rviz" pkg="rviz" type="rviz" 
                args="-f $(arg rviz_frame) -d $(arg rvizconfig)"
                ns="$(arg robot_name)">
            <remap from="/clicked_point" to="clicked_point"/>
            <remap from="/initialpose" to="initialpose"/>
            <remap from="/move_base_simple/goal" to="move_base_simple/goal"/>
        </node>
    </group>
</launch>